src/main/java/frc5124/robot2020/subsystems/Shooter.java

```diff
@@ -2,160 +2,73 @@
-        // Put code here to be run every loop
-        // read PID coefficients from SmartDashboard
-        double p = SmartDashboard.getNumber("P Gain", 0);
-        double i = SmartDashboard.getNumber("I Gain", 0);
-        double d = SmartDashboard.getNumber("D Gain", 0);
-        double iz = SmartDashboard.getNumber("I Zone", 0);
-        double ff = SmartDashboard.getNumber("Feed Forward", 0);
-        double max = SmartDashboard.getNumber("Max Output", 0);
-        double min = SmartDashboard.getNumber("Min Output", 0);
-
-        double setPoint = SmartDashboard.getNumber("SetPoint", 0);
-
-        // if PID coefficients on SmartDashboard have changed, write new values to controller
-        if((p != kP)) { m_pidController.setP(p); kP = p; }
-        if((i != kI)) { m_pidController.setI(i); kI = i; }
-        if((d != kD)) { m_pidController.setD(d); kD = d; }
-        if((iz != kIz)) { m_pidController.setIZone(iz); kIz = iz; }
-        if((ff != kFF)) { m_pidController.setFF(ff); kFF = ff; }
-        if((max != kMaxOutput) || (min != kMinOutput)) { 
-            m_pidController.setOutputRange(min, max); 
-            kMinOutput = min; 
-            kMaxOutput = max; 
-        }
-
-        /**
-         * PIDController objects are commanded to a set point using the 
-         * SetReference() method.
-         * 
-         * The first parameter is the value of the set point, whose units vary
-         * depending on the control type set in the second parameter.
-         * 
-         * The second parameter is the control type can be set to one of four 
-         * parameters:
-         *  com.revrobotics.ControlType.kDutyCycle
-         *  com.revrobotics.ControlType.kPosition
-         *  com.revrobotics.ControlType.kVelocity
-         *  com.revrobotics.ControlType.kVoltage
-         */
-
-        m_pidController.setReference(setPoint, ControlType.kVelocity);
+/**
+ * 
+ * WARNING 
+ * Untuned
+ * 
+ * 
+ * 
+ */
+public class Shooter implements Subsystem {
+  private CANSparkMax shootMotorFollower = new CANSparkMax(RobotMap.ShooterMap.shootFollowerCanID, MotorType.kBrushless);
+  private CANSparkMax shootMotorLeader = new CANSparkMax(RobotMap.ShooterMap.shootLeaderCanID, MotorType.kBrushless);
+  private CANPIDController shootController; 
+  private double currentVelocity;
+ 
 
-        SmartDashboard.putNumber("ProcessVariable", m_encoder.getVelocity());
+  
+  public Shooter() {
+    shootMotorFollower.follow(shootMotorLeader);
+    shootController = shootMotorLeader.getPIDController();
+    shootController.setD(RobotMap.ShooterMap.Kd);
+    shootController.setFF(RobotMap.ShooterMap.Kf);
+    shootController.setP(RobotMap.ShooterMap.Kp);
+    shootController.setI(RobotMap.ShooterMap.Ki);
+    shootController.setOutputRange(-RobotMap.ShooterMap.maxVelocity, RobotMap.ShooterMap.maxVelocity);
+  }
+
+  /**
+   * @param targetVelocity desired ft/s [advise 30 or 0]
+   */
+  public void setTargetVelocity(double targetVelocity) {
```

Comment:
I don't like the idea of passing the velocity value from outside the subsystem, but I understand what you're trying to do here to eliminate the stop method. the command could pass in a constant or an enum to work this way.